### ISSUE TITLE
Reply-To: is ignored in mew-config-alist has mail-domain setting.

### ISSUE THREAD

**Issue by ryoon** [kazu-yamamoto/Mew#25]
I use Mew 6.5 from released tar ball.
Reaply-To of e-mail is ignored when mew-config-alist has mail-domain setting in ~/.mew.

My ~/.emacs is here.
(setq load-path
  (append (list
                (expand-file-name "/opt/elisp/mew-6.5")
           )
           load-path))

(autoload 'mew "mew" nil t)
(autoload 'mew-send "mew" nil t)

And my ~/.mew is here.
(setq mew-config-alist
  '(
    (default
      (user "ryo_on")
      (name "Ryo ONODERA")
      (mail-domain "yk.rim.or.jp")
    )
  )
)

**Comment by kazu-yamamoto** [comment]
Twelve years late, sorry. This was measured rather than reasoned about.

## `mail-domain` is not the cause

Your configuration was reproduced as posted, and then with the
`mail-domain` line taken out, changing nothing else:

```
== with mail-domain, as reported ==
  addresses taken to be mine => ("^ryo_on$" "^ryo_on@yk\\.rim\\.or\\.jp$")
  is From:ryoon@NetBSD.org me? => nil
  To: of the reply => ("gnats-bugs@NetBSD.org" "ryoon@NetBSD.org")

== without mail-domain ==
  addresses taken to be mine => ("^ryo_on$" "^ryo_on@kazu-macbook-air\\.local$")
  is From:ryoon@NetBSD.org me? => nil
  To: of the reply => ("gnats-bugs@NetBSD.org" "ryoon@NetBSD.org")
```

`Reply-To:` is used either way. Setting `mail-domain` does not change
the outcome.

## What was actually happening

When the `From:` is taken to be one of your own addresses, Mew uses
`mew-reply-fromme-alist`, whose default is

```elisp
'((t
   ("To:" "To:")
   ("Cc:" "Cc:")
   ("Newsgroups:" "Newsgroups:")))
```

and which never looks at `Reply-To:` at all. Going down that path with
your message gives

```
  is From:ryo_on@yk.rim.or.jp me? => t
  To: of the reply => ("pkg-manager@NetBSD.org" "gnats-admin@NetBSD.org")
  Cc: of the reply => nil
```

which is exactly the header you posted. So your Mew was treating
`ryoon@NetBSD.org` as an address of yours.

## Why it would do that

Whether an address is yours is decided in
`mew-get-my-address-regex-list` from three things and no others:
`(mew-user)`, `(mew-mail-address)` and `mew-mail-address-list`.
`mail-domain` reaches this only through the second, and can never make
anything but `user@mail-domain` yours.

With the configuration you posted, `ryoon@NetBSD.org` is not yours. So
there was presumably something else in your real `.mew` which is not in
the report: `mew-mail-address-list`, or another case with a different
`user` or `mail-domain`.

@tats's last note fits here too: `mew-regex-my-address-list` is built
once, by `mew-init`, so a configuration changed without restarting
Emacs leaves the old set of addresses in place.

## In short

Not a bug, but the way a reply to your own message is treated.
@tats's `mew-reply-fromme-alist` setting is the way to have `Reply-To:`
honoured there, and it is the right answer to the original question.

Closing. Thank you for the report, and sorry it took this long.
